src/kvstore/raftex/Host.cpp

```diff
@@ -190,19 +192,24 @@ void Host::appendLogsInternal(folly::EventBase* eb,
             {
                 std::lock_guard<std::mutex> g(self->lock_);
                 self->setResponse(r);
-                self->lastLogIdSent_ = self->logIdToSend_;
+                self->lastLogIdSent_ = self->logIdToSend_ - 1;
```

Comment:
Not quite understanding why `logIdToSend_ - 1`

Reply:
It is used for avoid infinite loop when retrying for errors.  For more detail,  let's talk about it offline,  or you could ask for @critical27